### PR TITLE
web: grpc typescript (WIP)

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: web/src/viewServiceClientPb.ts [hunk 1]
```diff
@@ -6,55 +6,53 @@
 
 // GENERATED CODE -- DO NOT EDIT!
 
+import * as grpcWeb from "grpc-web"
 
-import * as grpcWeb from 'grpc-web';
+// NOTE(dmiller): why do we have to delete this?
+//import * as google_api_annotations_pb from '../../google/api/annotations_pb';
```
Comment (jazzdan) [inline web/src/viewServiceClientPb.ts]: It seems insistent on generating this, but it's never used and that's an error.

File: web/package.json [hunk 2]
```diff
@@ -16,6 +16,8 @@
     "@types/react-timeago": "^4.1.0",
     "@types/styled-components": "^4.1.19",
     "ansi-to-react": "^5.0.0",
+    "google-protobuf": "^3.10.0",
```
Comment (jazzdan) [inline web/package.json]: Do we need both of these?

File: web/src/HUD.tsx [hunk 3]
```diff
@@ -110,6 +112,14 @@ class HUD extends Component<HudProps, HudState> {
       // we don't want to run any bootstrapping code in the test environment
       return
     }
+    // TODO(dmiller): remove this
```
Comment (jazzdan) [inline web/src/HUD.tsx]: This is just some test code to see it working end-to-end.

File: web/src/view_pb.js [hunk 4]
```diff
@@ -0,0 +1,956 @@
+/* eslint-disable */
+// NOTE(dmiller): ^I have to add this
```
Comment (jazzdan) [inline web/src/view_pb.js]: Needed to add this

**Comment by jazzdan** [comment]
Another problem: the GRPC server responds on http://localhost:10350/api/proto/view but the generated code is making a request to http://localhost:10350/webview.ViewService/GetView

**Comment by nicks** [comment]
hmmm...let's talk about this tomorrow. in TFT, the standard pattern was to add an annotation to the protobuf that declared the url path, so i'm surprised that's not working here

File: toast.yml [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
 tasks:
   proto:
     input_paths:
       - pkg/webview/view.proto
     output_paths:
       - pkg/webview/view.pb.go
       - pkg/webview/view.pb.gw.go
-    command: protoc -I. -I/usr/include -I"$(go env GOPATH)"/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis --grpc-gateway_out=logtostderr=true:. --go_out=plugins=grpc,paths=source_relative:. pkg/webview/view.proto && goimports -local github.com/windmilleng/tft -w pkg/webview/view.pb.go
+      - web/src/
+    command: mkdir -p web/src && protoc -I. -I/usr/include -I"$(go env GOPATH)"/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis --grpc-gateway_out=logtostderr=true:. --go_out=plugins=grpc,paths=source_relative:. --js_out=import_style=commonjs:web/src --grpc-web_out=import_style=typescript,mode=grpcwebtext:web/src pkg/webview/view.proto && goimports -local github.com/windmilleng/tft -w pkg/webview/view.pb.go && mv web/src/Pkg/webview/* web/src && mv web/src/pkg/webview/* web/src && rm -rf web/src/Pkg && rm -rf web/src/pkg
```
Comment (nicks) [inline toast.yml]: can you move this into a shell script? this is becoming unreadable

Comment (nicks) [reply toast.yml]: also, i don't totally understand why you have both import_style=commonjs and import_style=typescript. i would expect those to be mutually exclusive

Comment (jazzdan) [reply toast.yml]: Yeah @nicks ... I was confused about that too. I was just following an example I found with GitHub search, but it seemed wrong to me.

File: web/src/view_pb.js [hunk 6]
```diff
@@ -0,0 +1,954 @@
...
+ * Generated by JsPbCodeGenerator.
+ * @param {Array=} opt_data Optional initial data array, typically from a
+ * server response, or constructed directly in Javascript. The array is used
+ * in place and becomes part of the constructed object. It is not cloned.
+ * If no data is provided, the constructed object will be empty, but still
+ * valid.
+ * @extends {jspb.Message}
+ * @constructor
+ */
+proto.webview.Resource = function(opt_data) {
```
[the remark this answers is not in the document]
Comment (jazzdan) [reply web/src/view_pb.js]: Yeah, so it's imported from the typescript code:

```
import {
  GetViewRequest,
  View} from './view_pb';
```

File: toast.yml (hunk 5 above)
Comment (jazzdan) [reply toast.yml]: but anyways I moved it in to a "shell script" :)

**Pull request closed without merging**